components/brave_shields/browser/tracking_protection_service.h

```diff
@@ -43,14 +47,51 @@ class TrackingProtectionService : public BaseLocalDataFilesObserver {
                         const base::FilePath& install_dir,
                         const std::string& manifest) override;
 
+#if BUILDFLAG(BRAVE_STP_ENABLED)
+  bool ShouldStoreState(HostContentSettingsMap* map,
```

Comment:
At the moment we utilize `TrackingProtectionService` at `OnBeforeURLRequest_AdBlockTPPreWork` (here we explicilty post from IO to UI and then back again).
Now we need it in MessageFilter, thus adding an extra lock because it is called on IO.

I think we should just accept that this service works entirely on IO thread and eliminate at once adding new lock and crossposting from UI.

Reply:
@iefremov - I think it makes sense to file a follow-up issue for this, and address this as a separate PR. Let me know what you think? cc: @bridiver

Reply:
Filed https://github.com/brave/brave-browser/issues/3189